utam-core/src/main/java/utam/core/framework/element/NavigationImpl.java

```diff
@@ -98,4 +105,59 @@ public void closeWindow(String targetUrl) {
   public void closeWindow() {
     this.driverAdapter.close();
   }
+
+  @Override
+  public int getWindowCount() {
+    return this.driverAdapter.getWindowHandles().size();
+  }
+
+  @Override
+  public Window currentWindow() {
+    return new WindowImpl(this.factory);
+  }
+
+  @Override
+  public void setupWaitForNewWindow() {
+    oldWindowCount = getWindowCount();
+    oldWindowHandles = new HashSet<>(this.driverAdapter.getWindowHandles());
+
+    this.setupForNewWindow = true;
+  }
+
+  @Override
+  public Window waitForNewWindow() {
+    if (!this.setupForNewWindow) {
```

Comment:
Keying off of `oldWindowHandles == null` here actually makes the code _more_ readable, imo, since it makes it explicit that references to that below won't throw `NullReferenceException`.